collectors/tc.plugin/plugin_tc.c

```diff
@@ -12,60 +12,48 @@
 #define TC_LINE_MAX 1024
 
 struct tc_class {
-    avl_t avl;
+    STRING *id;
+    STRING *name;
+    STRING *leafid;
+    STRING *parentid;
+
+    bool hasparent;
+    bool isleaf;
+    bool isqdisc;
+    bool render;
+    bool name_updated;
+    bool updated;
 
-    char *id;
-    uint32_t hash;
-
-    char *name;
-
-    char *leafid;
-    uint32_t leaf_hash;
-
-    char *parentid;
-    uint32_t parent_hash;
-
-    char hasparent;
-    char isleaf;
-    char isqdisc;
-    char render;
+    int  unupdated; // the number of times, this has been found un-updated
 
     unsigned long long bytes;
     unsigned long long packets;
     unsigned long long dropped;
-    unsigned long long overlimits;
-    unsigned long long requeues;
-    unsigned long long lended;
-    unsigned long long borrowed;
-    unsigned long long giants;
     unsigned long long tokens;
     unsigned long long ctokens;
 
+    //unsigned long long overlimits;
+    //unsigned long long requeues;
+    //unsigned long long lended;
+    //unsigned long long borrowed;
+    //unsigned long long giants;
+
     RRDDIM *rd_bytes;
     RRDDIM *rd_packets;
     RRDDIM *rd_dropped;
     RRDDIM *rd_tokens;
     RRDDIM *rd_ctokens;
-
-    char name_updated;
-    char updated;   // updated bytes
-    int  unupdated; // the number of times, this has been found un-updated
-
-    struct tc_class *next;
-    struct tc_class *prev;
 };
 
 struct tc_device {
-    avl_t avl;
-
-    char *id;
+    STRING *id;
     uint32_t hash;
```

Comment:
We don't use it anymore.

Reply:
I am confused. What we don't use? `STRING *id` on `struct tc_class`?

Reply:
Oh! `hash`! You are right!